### PR TITLE
Fix for an issue where initializing an (unintentional) empty model causes a push onto the Summernote History stack.

### PR DESCRIPTION
**Pull request by nwpan**
As per my issue post:

Hey all -- I've ran into a bug that affects the the first "undo" of Summernote. When you trigger an "undo" upon loading, the text gets set to a blank value. This affects the History stack of Summernote, causing a strange behaviour. I'm proposing this PR as a fix.

For a example, I've provided two scripts below... First is the original directive (loading Angular Summernote from CDN). The second is my modified fix, proposed at the above PR. Please see here:
* http://jsfiddle.net/kauypdyb/3/
* http://jsfiddle.net/x9ra1x7r/3/

Steps to reproduce:
1. Click the [first link](http://jsfiddle.net/kauypdyb/3/).
2. Press either CTRL+Z or press the backarrow undo button within the toolbar.
3. Notice that the text field becomes blank.

This issue affects when you select a text range, then **bold the text**, then press undo. In the native demo (without Angular), you are also not able to undo to a blank space. 

Specifically, this problem has to do with using the [transclude: 'element']: [here](https://github.com/summernote/angular-summernote/blob/master/src/angular-summernote.js#L133). As per this StackOverflow discussion ([here](http://stackoverflow.com/questions/18449743/angularjs-when-to-use-transclude-true-and-transclude-element)), we cannot have access to the scope because it hasn't been created yet. Specifically in the link function, we're passing an empty model into this function. Then we continue and activate the SummernoteController.

Instead, we need to yield and wait until the scope gets created. In my proposed change, I have checked if there is a model provided (a blank one), then watch for a change. I've also added a unit test to check when the Summernote element's HTML is blank. Please let me know what you guys think, or if there's a better approach to this.

Cheers,

Nick
:coffee:

### CODE REVIEW

**Comment by outsideris** [comment]
@nwpan Thank you for great PR. I can understand the issue easily because of you explanation. I'm reviewing your PR and consider your approach.

**Comment by outsideris** [comment]
Could you exclude all files in `dist/` folder? I want to update it only when new version released.

File: dist/angular-summernote.js [hunk 1]
```diff
@@ -153,14 +153,25 @@ angular.module('summernote', [])
...
-            ngModel = ctrls[1];
-
-        transclude(scope, function(clone, scope) {
-          // to prevent binding to angular scope (It require `tranclude: 'element'`)
-          element.append(clone.html());
-        });
+          ngModel = ctrls[1];
 
-        summernoteController.activate(scope, element, ngModel);
+          if (!ngModel)
```
Comment (outsideris) [inline dist/angular-summernote.js]: Could you `if-else` statement as below form?

```
if (!ngModel) {
   ...
} else {
   ...
}
```

File: src/angular-summernote.js [hunk 2]
```diff
@@ -152,14 +152,25 @@ angular.module('summernote', [])
...
-            ngModel = ctrls[1];
-
-        transclude(scope, function(clone, scope) {
-          // to prevent binding to angular scope (It require `tranclude: 'element'`)
-          element.append(clone.html());
-        });
+          ngModel = ctrls[1];
 
-        summernoteController.activate(scope, element, ngModel);
+          if (!ngModel)
```
Comment (outsideris) [inline src/angular-summernote.js]: Could you `if-else` statement as below form?

```
if (!ngModel) {
   ...
} else {
   ...
}
```

File: src/angular-summernote.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ angular.module('summernote', [])
           removedMedia.attrs[attr.name] = attr.value;
         });
         $scope.mediaDelete({target: removedMedia});
-      }
+    };
```
Comment (outsideris) [inline src/angular-summernote.js]: Please keep indentation.

**Comment by nwpan** [comment]
Thanks for reviewing the PR @outsideris -- I've addressed the issues you've raised. Cheers!

**Comment by outsideris** [comment]
@nwpan Great! Thank you for your contribution.

**Pull request merged**